.release-notes/4182.md

```diff
@@ -0,0 +1,29 @@
+## Enhance checking for identity comparison with new objects
+
+An identity comparison between a new object and something else is always going to fail.
+
+This was correctly spotted by the compiler when using sugared constructor calls.
```

Comment:
This paragraph can be removed. I think the release notes are much better without it.